File: source/camera.py
__name__ = "camera"

from .pixeldata import PixelMetaData
import os

class Camera:
    image = [[]]
    def __init__(self, res_width, res_height):
        self.resolutionWidth = res_width
        self.resolutionHeight = res_height
        self.image = [[PixelMetaData() for _ in range(res_height)] for _ in range(res_width)]
    
    def print(self):
        for i in range(self.resolutionHeight):
            for j in range(self.resolutionWidth):
                phstr = '['+str(self.image[i][j].x)+',' +str(self.image[i][j].y)+']'
                os.sys.stdout.write(phstr)
            print()

    

File: source/camera.cpp
#include "camera.h"

Camera::Camera(int resWidth, int resHeight):canvas(std::map<int, std::map<int, PixelMetaData>>()){
    frameWidth = resWidth;
    frameHeight = resHeight;
    for(int row = 0; row < frameHeight; row++){
        canvas[row] = std::map<int,PixelMetaData>();
        for(int col = 0; col < frameWidth; col++){
            canvas[row][col] = PixelMetaData();
        }
    }
}

void Camera::print(){
    for(int row = 0; row < frameHeight; row++){
        for(int col = 0; col < frameWidth; col++){
            std::cout << "[" << canvas[row][col].x << ","
                        << canvas[row][col].y << "]";
        }
        std::cout << std::endl;
    }
}